quint/src/cliCommands.ts

```diff
@@ -554,7 +554,36 @@ export async function verifySpec(prev: TypecheckedStage): Promise<CLIProcedure<V
   const args = verifying.args
   // TODO error handing for file reads and deserde
   const loadedConfig = args.apalacheConfig ? JSON.parse(readFileSync(args.apalacheConfig, 'utf-8')) : {}
-  const parsedSpec = jsonStringOfOutputStage(pickOutputStage(prev))
+
+  // Flatten modules, replacing instances, imports and exports with their definitions
+  const [analysisErrors, analysisOutput] = analyzeModules(verifying.table, verifying.modules)
+
+  if (analysisErrors.length > 0) {
+    const errorLocator = mkErrorMessage(verifying.sourceMap)
+    const errors = Array.from(analysisErrors, errorLocator)
+    return cliErr('verification failed', { ...verifying, errors })
+  }
```

Comment:
I'm not sure if this is ready for review yet since it is in draft mode, but anyway, here's one comment:

You shouldn't need to run analysis again, as `prev` is a `TypecheckedStage` and therefore should have the analysis output in it's properties. irrc, there's no `analysisOutput` property itself, but there should be `types`, `effects` and `modes`, so you can construct the `AnalysisOutput` object from those.

Reply:
I propose we first make things work in this PR and then refactor them to make code reuse

Reply:
I don't think there's any reuse to be obtained here besides avoiding rerunning analysis.

Reply:
> I had naively assumed flattening would already be taken care of in the processing pipeline up to this point.

Currently we just call flattening at the REPL/compiler since we only need it for that stage. And it is not possible to just pre-process everything before the compiler because of its interactivity (i.e. if an import statement is written into the REPL, it will need to re-process a lot of stuff, including reflattening).

So this:

> I don't think there's any reuse to be obtained here besides avoiding rerunning analysis.

is correct.

On the topic of further refactor after this PR, I was thinking about introducing some sort of structure to aggregate results between "compilation" (parsing/analysis/flattening/compilation) stages, and I only now see this is pretty much what we have with these CLI stages. I wonder whether it's a good idea to use these stages beyond the CLI implementation, and then, as Shon mentions in his [comment](https://github.com/informalsystems/quint/pull/944#discussion_r1230986373) about arguments, avoid having a lot of arguments being propagated and use a single object with the aggregated information. You'll see in my next PR how cumbersome managing all those arguments can be.

Reply:
Straightened out the implementation in [05f7c0b](https://github.com/informalsystems/quint/pull/944/commits/05f7c0b8d35e7dcb5402447d3ca4d72be6b238f2)